File: java/sourceafis/src/sourceafis/extraction/model/BranchMinutiaRemover.java
/**
 * @author Veaceslav Dubenco
 * @since 20.10.2012
 */
package sourceafis.extraction.model;

import sourceafis.general.DetailLogger;

/**
 * 
 */
public class BranchMinutiaRemover implements ISkeletonFilter {

	public DetailLogger.Hook Logger = DetailLogger.off;

	@Override
	public void Filter(SkeletonBuilder skeleton) {
		for (SkeletonBuilderMinutia minutia : skeleton.getMinutiae()) {
			if (minutia.getRidges().size() > 2)
				minutia.Valid = false;
		}
		Logger.log(skeleton);
	}

}

File: SourceAFIS/Extraction/Model/BranchMinutiaRemover.cs
using System;
using System.Collections.Generic;
using System.Text;
using SourceAFIS.General;

namespace SourceAFIS.Extraction.Model
{
    public sealed class BranchMinutiaRemover : ISkeletonFilter
    {
        public void Filter(SkeletonBuilder skeleton)
        {
            foreach (SkeletonBuilder.Minutia minutia in skeleton.Minutiae)
            {
                if (minutia.Ridges.Count > 2)
                    minutia.Valid = false;
            }
        }
    }
}
